### ISSUE TITLE
sql: duplicates in SHOW QUERIES output

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/cockroach#20349]
Report from Gitter:

- v1.1.2
- 4 node cluster

`SHOW QUERIES` output:

```
| 14fb94720a9dfc390000000000000001 | 1 | root | 2017-11-29 14:21:57.43735+00:00 | UPSERT abc | 141.28.74.103:38602 | PostgreSQL JDBC Driver | false | executing |
| 14fb94744f6b37970000000000000001 | 1 | root | 2017-11-29 14:21:57.43735+00:00 | UPSERT abc | 141.28.74.103:38602 | PostgreSQL JDBC Driver | false | executing |
| 14fb9475a1dff9f00000000000000002 | 2 | root | 2017-11-29 14:22:02.452937+00:00 | UPSERT xyz | 141.28.74.103:60366 | PostgreSQL JDBC Driver | false | executing |
| 14fb9473358e9d510000000000000002 | 2 | root | 2017-11-29 14:22:02.452937+00:00 | UPSERT xyz | 141.28.74.103:60366 | PostgreSQL JDBC Driver | false | executing |
```

`abc` and `xyz` are different queries. User says each is sent only one time, so why are they each listed twice?

User also says that `SHOW SESSIONS` shows two session with `ROLLBACK TRANSACTION TO SAVEPOINT COCKROACH_RESTART`

**Comment by jseldess** [comment]
@jordanlewis, since you started looking into this already, assigning you initially. Feel free to reassign if that's better.

**Comment by dt** [comment]
Just saw this too. I started a long-running update (65106 rows) and was keeping an eye on `SHOW QUERIES` to see if others were piling up. After ~3min I saw a second copy of my update appear, exactly the same as the first (including start time), except with a different query ID. A third copy appeared by 8 minutes in and a fourth a few minutes later.

**Comment by knz** [comment]
This looks like different query IDs are generated for txn retries.

**Comment by andzdroid** [comment]
I reported this issue originally. I'm not sure what fixed it but I haven't had this problem anymore since upgrading from 1.1.4 to the first v2 alpha.

**Comment by leomkkwan** [comment]
I still observe this on v2.0-alpha.20171218.

**Comment by jordanlewis** [comment]
I think this is explained by #22574, which has been fixed by the new conn_executor on release-2.0.
